slab/transfer.go

```diff
@@ -80,59 +80,25 @@ func parallelUploadSlab(shards [][]byte, hosts []Host) ([]Sector, error) {
 	return sectors, nil
 }
 
-// UploadSlabs uploads slabs read from the provided Reader.
-func UploadSlabs(r io.Reader, m, n uint8, hosts []Host) ([]Slab, error) {
+// UploadSlab uploads a slab.
+func UploadSlab(r io.Reader, m, n uint8, hosts []Host) (Slab, error) {
 	buf := make([]byte, int(m)*rhpv2.SectorSize)
 	shards := make([][]byte, n)
```

Comment:
you could allocate the `shards` slice after doing `io.ReadFull` saving the allocation in case of an error

Reply:
unlikely in practice, not worth the readability cost imo.

Down the line, we'll probably be passing in these buffers explicitly anyway, or fetching them from a pool or something